### PR TITLE
New years fireworks

### PR DESCRIPTION
**Pull request by MadmanMartian**
adds a firework event admins can trigger, fires fireworks at the station that do no damage, but look 'pretty'

The sprite does not look very nice, but it works

firework

Uses color matrix to change it to strange colors, same is done for the firework sparkle

- [x] Fix the firework sparkle from not dying properly.

### CODE REVIEW

File: code/game/gamemodes/meteor/meteors.dm [hunk 1]
```diff
@@ -478,3 +478,59 @@ var/list/blob_candidates = list()
...
+		list(1, 1, 0, 1),
+		list(0, 1, 1, 1),
+		list(1, 0, 1, 1)
+	)
+	var/list/cl = list(0.1,0.1,0.1,0.1)
+	for(var/x = 1 to 4)
+		cl = pick(colors) + cl
+	color = cl
+
+/obj/item/projectile/meteor/to_bump(atom/A)
```
Comment (MadmanMartian) [inline code/game/gamemodes/meteor/meteors.dm]: aw fuck

File: code/game/gamemodes/meteor/meteors.dm [hunk 2]
```diff
@@ -478,3 +478,67 @@ var/list/blob_candidates = list()
...
+		list(0, 0, 1, 1),
+		list(1, 1, 0, 1),
+		list(0, 1, 1, 1),
+		list(1, 0, 1, 1)
+	)
+	var/list/cl = list(0,0,0,0)
+	for(var/x = 1 to 4)
+		cl = pick(colors) + cl
+	color = cl
+	set_light(12,12,cl)
```
Comment (MadmanMartian) [inline code/game/gamemodes/meteor/meteors.dm]: This does not work.

File: code/game/gamemodes/meteor/meteors.dm [hunk 3]
```diff
@@ -478,3 +478,67 @@ var/list/blob_candidates = list()
...
+
+	var/matrix/M = src.transform
+	if(!istype(M))
+		M = matrix()
+	var/scaleby = rand(1,4)
+	M.Scale(scaleby, scaleby)
+
+	animate(src, transform = M, alpha = 0, time = 12)
```
Comment (MadmanMartian) [inline code/game/gamemodes/meteor/meteors.dm]: This does not work, does not fade out

File: code/game/gamemodes/meteor/meteors.dm (hunk 2 above)
Comment (PJB3005) [reply code/game/gamemodes/meteor/meteors.dm]: Lighting can't do color matrices for light color. Use RGB colors only.

File: code/game/gamemodes/meteor/meteors.dm (hunk 3 above)
Comment (PJB3005) [reply code/game/gamemodes/meteor/meteors.dm]: Weird. Looks fine to me.

File: code/game/gamemodes/meteor/meteors.dm (hunk 2 above)
Comment (MadmanMartian) [reply code/game/gamemodes/meteor/meteors.dm]: anything in IconProcs.dm that can convert from matrices to RGB?

Comment (MadmanMartian) [reply code/game/gamemodes/meteor/meteors.dm]: RGBtoHSV looks promising, testing now

**Comment by ShiftyRail** [comment]
>tfw Travis kills new year celebration

**Pull request merged**